src/interpreter/mod.rs

```diff
@@ -123,6 +125,19 @@ impl<'txin> Interpreter<'txin> {
         }
     }
 
+    /// Whether this is a pre-segwit spend
+    pub fn is_legacy(&self) -> bool {
```

Comment:
Should we also have a function for segwitv0? Should this be a part of `Descriptor` struct?

Reply:
Maybe descriptor is not the place, but interpreter.is_legacy() does not sound right.

Reply:
We can rename the interpreter...but this is the object that you get when you try to take a txin and analyze it using Miniscript. Determining legacy or not seems perfectly reasonable.